src/actions/mod.rs

```diff
@@ -156,6 +156,20 @@ impl ActionHandler {
         }
     }
 
+    pub fn find_impls(&self, id: usize, params: TextDocumentPositionParams, out: &Output) {
+        let span = self.convert_pos_to_span(&params.text_document, params.position);
+        let type_id = self.analysis.id(&span).unwrap();
+        // FIXME send failure
+        let res = self.analysis
+            .find_impls(type_id)
+            .unwrap_or(vec![])
+            .into_iter()
+            .map(|x| ls_util::rls_to_location(&x))
+            .collect();
```

Comment:
Could we spawn a thread with timeout to do this please? We do that for the other queries (e.g., find all refs) and I think we should do it here too).